pkg/linters/components/pkgconfigusage/pkgconfigusage.go

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package pkgconfigusage provides a linter for ensuring pkg/config is not used inside comp folder
+package pkgconfigusage
+
+import (
+	"strings"
+
+	"github.com/golangci/plugin-module-register/register"
+	"golang.org/x/tools/go/analysis"
+)
+
+// We can replace during tests
+var componentPath = "github.com/DataDog/datadog-agent/comp"
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** Should `componentPath` have a trailing `/` ? That way it wouldn't risk mismatches with eg. `github.com/DataDog/datadog-agent/comparison`‏ if that ever exists (unlikely to happen but you never know)
```suggestion
var componentPath = "github.com/DataDog/datadog-agent/comp/"
```

Reply:
Super fair point. I will add the trailing dash